libs/langchain/langchain/chat_models/ollama.py

```diff
@@ -0,0 +1,122 @@
+import json
+from typing import Any, Iterator, List, Optional
+
+from langchain.callbacks.manager import (
+    CallbackManagerForLLMRun,
+)
+from langchain.chat_models.base import BaseChatModel
+from langchain.llms.ollama import _OllamaCommon
+from langchain.schema import ChatResult
+from langchain.schema.messages import (
+    AIMessage,
+    AIMessageChunk,
+    BaseMessage,
+    ChatMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain.schema.output import ChatGeneration, ChatGenerationChunk
+
+
+def _stream_response_to_chat_generation_chunk(
+    stream_response: str,
+) -> ChatGenerationChunk:
+    """Convert a stream response to a generation chunk."""
+    parsed_response = json.loads(stream_response)
+    generation_info = parsed_response if parsed_response.get("done") is True else None
+    return ChatGenerationChunk(
+        message=AIMessageChunk(content=parsed_response.get("response", "")),
+        generation_info=generation_info,
+    )
+
+
+class ChatOllama(BaseChatModel, _OllamaCommon):
+    """Ollama locally runs large language models.
+
+    To use, follow the instructions at https://ollama.ai/.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.chat_models import ChatOllama
+            ollama = ChatOllama(model="llama2")
+    """
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "ollama-chat"
+
+    @property
+    def lc_serializable(self) -> bool:
+        return True
+
+    def _format_message_as_text(self, message: BaseMessage) -> str:
```

Comment:
any reason to make this function something that can be passed in? eg is it the case that for *any* model accessed through llama you'd want to use [INST] tags, or would folks want to customize for some models?

Reply:
Yeah definitely but I think we can add it later as more models become available